R/enum.R

```diff
@@ -0,0 +1,25 @@
+
+#' Create an enumeration to be used instead of arbitraty values in code.
+#'
+#' @param enumValues
+#'
+#' @return the Enum created
+#' @examples
+#'
+#' # Without predefined values
+#' Color <- enum(c("Red", "Blue", "Green"))
+#' myColor <- Color$Red
+#'
+#' # With predefined values
+#' Symbol <- enum(c(Diamond=1, Triangle=2, Circle=2))
+#' mySymbole <- Symbol$Diamond
+enum <- function(enumValues) {

```

Comment:
We will need to experiment with this but this is working quite nicely so far .

Reply:
This is not without names, this is without values
I could create enum(c(Diamond, Triangle, Circle)) and this would work